src/core/server/elasticsearch/integration_tests/client.test.ts

```diff
@@ -55,3 +63,67 @@ describe('elasticsearch clients', () => {
     expect(resp.headers!.warning).toMatch('system indices');
   });
 });
+
+function createFakeElasticsearchServer() {
+  const firstRequestReceived$ = new Subject<boolean>();
+
+  const requestHandler = jest.fn((req, res) => {
+    // Reply with a 200 and empty response by default (intentionally malformed response)
+    res.writeHead(200);
+    res.end();
+
+    firstRequestReceived$.next(true);
+  });
+
+  const server = http.createServer(requestHandler);
+  server.listen(esTestConfig.getPort());
+  return { server, requestHandler, firstRequestReceived$ };
+}
+
+describe('fake elasticsearch', () => {
+  let esServer: http.Server;
+  let requestHandler: jest.Mock;
+  let kibanaServer: Root;
+
+  beforeAll(async () => {
+    kibanaServer = createRootWithCorePlugins(
+      {
+        logging: { silent: false },
```

Comment:
Is it necessary for the test?

Reply:
I think that I was testing stuff and left it. I'll remove it. 👍

Reply:
Removed config left-overs of my manual tests.